src/frontend/screens/Quests/components/QuestDetailsViewPlay/index.tsx

```diff
@@ -12,222 +12,263 @@ import { useNavigate } from 'react-router-dom'
 
 export interface QuestDetailsViewPlayWrapperProps {
   selectedQuestId: number | null
 }
 
-export function QuestDetailsViewPlayWrapper({
-  selectedQuestId
-}: QuestDetailsViewPlayWrapperProps) {
-  const navigate = useNavigate()
-  const { t } = useTranslation()
-  const [collapseIsOpen, setCollapseIsOpen] = useState(false)
-
-  const questResult = useGetQuest(selectedQuestId)
-  const questMeta = questResult.data.data
-  const questPlayStreakResult = useGetUserPlayStreak(selectedQuestId)
-  const questPlayStreakData = questPlayStreakResult.data.data
-
-  const rewardsQuery = useGetRewards(selectedQuestId)
-  const questRewards = rewardsQuery.data.data
-
-  const getSteamGameResult = useGetSteamGame(
-    questMeta?.eligibility?.steam_games ?? []
-  )
-
-  const navigateToGame = useMutation({
-    mutationFn: async (appName: string) => {
-      const { appName: epicAppName, epicListingUrl } = await fetchEpicListing(
-        appName
-      )
+export const QuestDetailsViewPlayWrapper = observer(
+  ({ selectedQuestId }: QuestDetailsViewPlayWrapperProps) => {
+    const navigate = useNavigate()
+    const { t } = useTranslation()
+    const flags = useFlags()
+    const questsWithExternalSync: number[] = flags.questsWithExternalSync
+    const [collapseIsOpen, setCollapseIsOpen] = useState(false)
 
-      let runner: Runner = 'hyperplay'
-      let name = appName
-      if (epicListingUrl) {
-        runner = 'legendary'
-        if (epicAppName) {
-          name = epicAppName
-        }
+    const questResult = useGetQuest(selectedQuestId)
+    const questMeta = questResult.data.data
+    const questPlayStreakResult = useGetUserPlayStreak(selectedQuestId)
+    const questPlayStreakData = questPlayStreakResult.data.data
+
+    const rewardsQuery = useGetRewards(selectedQuestId)
+    const questRewards = rewardsQuery.data.data
+
+    const getSteamGameResult = useGetSteamGame(
+      questMeta?.eligibility?.steam_games ?? []
+    )
+
+    const { syncPlayStreakWithExternalSource } =
+      useSyncPlayStreakWithExternalSource({
+        refreshPlayStreak: questPlayStreakResult.invalidateQuery
+      })
+
+    const handleSyncPlayStreak = async () => {
+      if (!questMeta) {
+        console.error('Quest meta is null')
+        window.api.logError('Quest meta is null')
+        return
       }
 
-      // check for gameinfo to see if it is on the library
-      return getGameInfo(name, runner)
-        .then((res) => {
-          if (!res) {
-            throw new Error('Game not found in library')
+      if (questsWithExternalSync.includes(questMeta.id)) {
+        syncPlayStreakWithExternalSource(questMeta.id)
+      } else {
+        await window.api.syncPlaySession(questMeta.project_id, 'hyperplay')
+        await questPlayStreakResult.invalidateQuery()
+        window.api.logInfo(`Synced play session for quest ${questMeta.id}`)
+        console.log(`Synced play session for quest ${questMeta.id}`)
+      }
```

Comment:
we should not be syncing the playsession from the quests page since play streaks can only be earned by playing the game, which is why we sync in the overlay quest details